src/components/Select/Select.stories.tsx

```diff
@@ -305,6 +305,19 @@ export const WithSelectedOption: StoryObj<typeof Select> = {
   },
 };
 
+/**
+ * Use the `by` option to determine the selection (when using objects for the value list).
+ * - The type comparison can be by a named key in the object `by={'id'}` or using a comparison function
+ *
+ * See: https://headlessui.com/v1/react/listbox#listbox
+ */
+export const WithSelectedBy: StoryObj<typeof Select> = {
+  args: {
+    ...WithSelectedOption.args,
```

Comment:
[non-blocking] It might be a more thorough test to use `defaultValue: {...exampleOptions[1]}` instead of inheriting `defaultValue: exampleOptions[1]` from `WithSelectedOption` since the purpose of `by` is to handle cases where the values are copies rather than the exact same objects.

Reply:
++ good call. I suspect it will work properly, but this will prevent some by-ref getting thru down the road